relay/abi.go

```diff
@@ -31,54 +31,43 @@ var ABI string
 
 // from `forge inspect Jackal abi`
 
-var (
-	eventABI abi.ABI
-	errABI   error
-
-	evmAddress  string
-	messageType string
-	cost        int64
-	relayedMsg  evmTypes.ExecuteMsg
-	factoryMsg  evmTypes.ExecuteFactoryMsg
-
-	errUnpack   error
-	errGenerate error
-)
+var eventABI abi.ABI
 
 func init() {
-	eventABI, errABI = abi.JSON(strings.NewReader(ABI))
+	e, errABI := abi.JSON(strings.NewReader(ABI))
 	if errABI != nil {
 		log.Fatalf("Failed to parse ABI: %v", errABI)
 	}
+	eventABI = e
 }
 
-func generatePostedFileMsg(w *wallet.Wallet, q *uploader.Queue, chainID uint64, event PostedFile) (err error) {
+func generatePostedFileMsg(w *wallet.Wallet, q *uploader.Queue, chainID uint64, event PostedFile) (string, *evmTypes.ExecuteMsg, int64, error) {
 	log.Printf("Event details: %+v", event)
-	evmAddress = event.From.String()
+	evmAddress := event.From.String()
 
 	merkleBase64, err := merkleToString(event.Merkle)
 	if err != nil {
```

Comment:
nit: log error here

Reply:
what's the log error sorry?

Reply:
i thought we should log the error you get from line 49 instead of just returning